src/ui/dialog/dialog.tsx

```diff
@@ -0,0 +1,68 @@
+import styles from './dialog.module.css'
+
+import React, { ReactElement, ReactNode } from 'react'
+import classNames from 'classnames'
+
+import { Icon } from '../icon'
+import { IModalBaseProps, Modal } from '../modal'
+
+export type TDialogWidth = 's' | 'm' | 'l'
+
+export interface IDialogProps
+  extends Omit<IModalBaseProps, 'transitionDuration'> {
+  width?: TDialogWidth
+  modalClassName?: string
+  transitionDuration?: number
+  className?: string
+  title?: ReactNode
+  actions?: ReactNode
+  children?: ReactNode
+}
+
+export const Dialog = ({
+  className,
+  modalClassName,
+  isOpen,
+  isKeepMounted,
+  width = 's',
+  transitionDuration = 300,
+  onClose,
+  title,
+  actions,
+  children,
+}: IDialogProps): ReactElement => {
+  return (
+    <Modal
+      className={modalClassName}
+      transitionDuration={transitionDuration}
+      isOpen={isOpen}
+      isKeepMounted={isKeepMounted}
+      onClose={onClose}
+      render={({ state }) => (
+        <div
+          className={classNames(
+            styles.dialog,
+            styles[`dialog_${state}`],
+            styles[`dialog_${width}`],
+            className,
+          )}
+          style={{
+            transitionDuration: `${transitionDuration}ms`,
+            visibility: state === 'exited' && !isOpen ? 'hidden' : undefined,
+          }}
+        >
+          <button
+            tabIndex={-1}
+            className={styles.dialog__close}
+            onClick={onClose}
+          >
+            <Icon name="Close" size={16} />
+          </button>
+          {title && <h2 className={styles.dialog__title}>{title}</h2>}
+          <div className={styles.dialog__content}>{children}</div>
+          {actions && <div className={styles.dialog__actions}>{actions}</div>}
+        </div>
```

Comment:
Ideally, I would like to see it like that:
<Modal {props}> -- to make something be opened in portal
    <Modal(Popup)Card {props}> -- default modal card component with header and footer
         <content> -- the actual content of the window
              ...some stuff
         </content>
    </Modal(Popup)Card>
</Modal>

Alternatively, <Modal> and <Modal(Popup)Card> could be wrapped with some <Dialog>, thus the structure will be:
<Dialog {props}> -- having Modal and Card within
    <content>
       ...some stuff
    </content>
</Dialog>

It would be easier to use.